### PR TITLE
feat: add PodMonitor for network-costs

### PR DESCRIPTION
**Pull request by onefatrabbit**
Added a switch for turning on a servicemonitor and a service for the network-costs daemonset in case you want to run a prometheus-operator.

### CODE REVIEW

**Comment by mbolt35** [comment]
@onefatrabbit Thanks for taking a crack at this!

My main concern here is that the `Service` definition will provide round-robin access to the network cost pods, correct? If so, instead of this behavior, we want each daemonset pod to be scraped by prometheus, not just one each scrape. Does that make sense? The prometheus config we use to do this is here: https://github.com/kubecost/cost-analyzer-helm-chart/blob/89cf47a36bef0707990ddd6cf9175bae9ef5d482/cost-analyzer/values.yaml#L147-L154

The `role: pod` ensures that each pod with the label is scraped.

**Comment by onefatrabbit** [comment]
@mbolt35 first of all, thanks a lot for pointing that out

you are absolutely right, scraping via the role: pod is the reasonable way. We might have opened this pull request a bit prematurely. And we totally overlooked that baked in relabel config, tbh.
The idea behind it was, that it would benefit the usage of a prometheus operator. So we have the whole cost-analyzer setup located in the cost analyzer helm chart, and don't need to extend our additionalScrapeConfigs file.

**Comment by dwbrown2** [comment]
@onefatrabbit did you find a solution here? let us know if we can help with this PR!

File: cost-analyzer/templates/cost-analyzer-network-costs-podmonitor-template.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{{- if .Values.networkCosts.podMonitor }}
+{{- if .Values.networkCosts.podMonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: {{ include "cost-analyzer.networkCostsName" . }}
+  labels:
+    {{ include "cost-analyzer.commonLabels" . | nindent 4 }}
+    {{- if .Values.networkCosts.podMonitor.additionalLabels }}
+    {{ toYaml .Values.networkCosts.e.additionalLabels | nindent 4 }}
```
[the remark this answers is not in the document]
Comment (mbolt35) [reply cost-analyzer/templates/cost-analyzer-network-costs-podmonitor-template.yaml]: Definitely looked like a typo, I've made some adjustments and merged in the most recent develop branch.

**Comment by mbolt35** [comment]
This looks good to me!

**Pull request merged**